hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -554,8 +554,7 @@ protected void postCommit(HoodieTable table, HoodieCommitMetadata metadata, Stri
    */
   protected void mayBeCleanAndArchive(HoodieTable table) {
     autoCleanOnCommit();
-    // reload table to that timeline reflects the clean commit
-    autoArchiveOnCommit(createTable(config, hadoopConf));
+    autoArchiveOnCommit(table);
```

Comment:
This is unnecessary, we just need to refresh the table metadata active timeline.